### ISSUE TITLE
Expecting value: line 1 column 1 (char 0)

### ISSUE THREAD

**Issue by mrplow** [owncloud/news-updater#1]
Debian Jessie
python3.4 installed, updater installed with pip, running owncloud-news-updater without any args prints the help menu, however when I point it to the updater.ini or run it with only my owncloud path results in the following:

```
root@5d991fc9f6e2:~# owncloud-news-updater -c /etc/owncloud/news/updater.ini 
2016-04-04 04:47:32,279 - ownCloud News Updater - ERROR - Expecting value: line 1 column 1 (char 0): Trying again in 30 seconds
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/owncloud_news_updater/updater.py", line 45, in run
    feeds = self.all_feeds()
  File "/usr/local/lib/python3.4/dist-packages/owncloud_news_updater/updater.py", line 198, in all_feeds
    return json.loads(feeds_json)['feeds']
  File "/usr/lib/python3.4/json/__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.4/json/decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.4/json/decoder.py", line 361, in raw_decode
    raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)
^CTraceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/owncloud_news_updater/updater.py", line 45, in run
    feeds = self.all_feeds()
  File "/usr/local/lib/python3.4/dist-packages/owncloud_news_updater/updater.py", line 198, in all_feeds
    return json.loads(feeds_json)['feeds']
  File "/usr/lib/python3.4/json/__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.4/json/decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.4/json/decoder.py", line 361, in raw_decode
    raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/owncloud-news-updater", line 9, in <module>
    load_entry_point('owncloud-news-updater==8.2.6', 'console_scripts', 'owncloud-news-updater')()
  File "/usr/local/lib/python3.4/dist-packages/owncloud_news_updater/application.py", line 131, in main
    updater.run()
  File "/usr/local/lib/python3.4/dist-packages/owncloud_news_updater/updater.py", line 71, in run
    time.sleep(30)
KeyboardInterrupt
```

**Comment by Bugsbane** [comment]
I'm getting this as well on Debian Jessie. A couple of notes, though:
1. There is no /etc/owncloud directory, despite being mentioned in the systemd service text.
2. The systemd service text also mentions the http user. On my owncloud install (which was done via Yunohost) all owncloud files are owned by the user owncloud and not the http user
3. When first installing owncloud news updater via pip3 it gave errors about requirements.txt not existing, but still said installation performed correctly. owncloud-news-updater does exist in /usr/bin

Tech specs:
Debian Jessie, fully updated
Nginx 1.6.2
Owncloud 9.0.1 installed via Yunohost
PHP 5.6.19

**Comment by BernhardPosselt** [comment]
Please rerun this with loglevel info. Probably the JSON output is broken

**Comment by BernhardPosselt** [comment]
@Bugsbane 
1. The systemd stuff turned out to be too platform specific, see https://github.com/owncloud/news-updater#running-the-updater-as-systemd-service
2. Further down it also mentions that this varies depending on your distro and that debian uses www-data
3. The requirements.txt is not needed anymore, found it in the MANIFEST.in file and removed it. Errors should be gone

**Comment by BernhardPosselt** [comment]
I've released 8.3.1 which fixes the warning and gives you better JSON error output.

**Comment by Bugsbane** [comment]
I was getting this again, but I figured out the problem.

The instructions say to run:

> owncloud-news-updater /path/to/owncloud

but this actually fails if you run it as any user other than the webserver user, so it really should say to test with:

> sudo -u YOUR-WEBSERVER-USER owncloud-news-updater /path/to/owncloud

or however you want to word "webserver user". Using the correct user instantly makes the problem disappear when testing.

This also needs to be made clearer in the systemd configuration file section (ie it assumes http as the correct user without explanation or clearly marking that this may need to be changed)

After figuring out those two parts of the docs, the news updater (and ownCloud news) are now working beautifully for me. Thanks for your work on this!

**Comment by BernhardPosselt** [comment]
Please read the readme again ;) it's described in the systemd section. The rest depends on your user which could be any user. Will add a hint to the updater command

**Comment by Bugsbane** [comment]
I did read the readme. The issue is that while in one section (of a rather long and complicated document) it mentions needing to use your webserver user, that in the specific text of another section where you tell people what command to use, that text fails unless you adapt it as I posted above.

Specifically, if people put in the command owncloud-news-updater /path/to/owncloud _as the readme states they should do_ then for many people that will fail. To get it to work, they have to realize that something written in another section of the readme means that they need to figure out how to adapt the command that the readme says should be used (and not just the username, but also to use sudo -u).

Is all the information available for someone dedicated to go through and figure it out if they're highly technically savvy. Sure. I did, even if it ended up needlessly costing me an hour or so to figure out.

Is it written in a way that clear, concise and will minimize people getting errors and bugging you needlessly?

Well, I can tell you that if you made the small change I suggested above, you wouldn't be having this conversation with me, and possibly mrplow right now. And that doesn't factor in other people who have gotten the same error, spent time trying to figure it out, possibly given up, and not bothered to raise an issue on GitHub.

Anyway, your call as to what you prefer. I just figured I'd spend a little of my time to save your time and your users.
